src/bin/communication_thread.rs

```diff
@@ -0,0 +1,65 @@
+//! Example on how to use a communication thread alongside with the GUI thread.
+//! 
+//! Tricks used here:
+//! - Use a queue to show data on the GUI.
+//! - Run an async function on the GUI event loop.
+//! - Use a seperate thread to handle incoming data and put it into a queue.
+
+extern crate futures;
+extern crate gio;
+extern crate glib;
+extern crate gtk;
+use futures::StreamExt;
+use gio::prelude::*;
+use gtk::prelude::*;
+use gtk::{ApplicationWindow, Label};
+use std::env::args;
+use std::thread;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.communication_thread"),
+        Default::default(),
+    )
+    .expect("Initialization failed...");
+    application.connect_activate(build_ui);
+    application.run(&args().collect::<Vec<_>>());
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = ApplicationWindow::new(application);
+    let label = Label::new(None);
+    window.add(&label);
+
+    // Create a channel between communication thread and main event loop:
+    let (sender, mut receiver) = futures::channel::mpsc::channel::<String>(1000);
+
+    // Spawn queue receive task on the main event loop:
+    let main_context = glib::MainContext::default();
+    let future = async move {
+        while let Some(item) = receiver.next().await {
```

Comment:
Like this using an async block over the main context channel has not that much advantage. They only really begin to shine if you run multiple async operations in them, i.e. have multiple awaits. Not sure what else to do here though :)

Reply:
Agreed, but I do not know what to do further and keep the example small at the same time. I'm hoping people familiar with async/await will know the advantage.

Reply:
Ok :)